server/controllers/users.js

```diff
@@ -282,3 +284,30 @@ export function getCallCompleteTwiml(req, res) {
     .set({ 'Content-Type': 'text/xml' })
     .send(callCompletedTwiml);
 }
+
+export function connectVolunteerToContact(req, res) {
+  const user_id = parseInt(req.params.id, 10);
+  const campaign_id = parseInt(req.params.campaign_id, 10);
+  const call_id = parseInt(req.params.call_id, 10);
+  return callsService.getCallById({ id: call_id })
+  .then((call) => {
+    const { contact_id } = call.attributes;
+    return contactsService.getContactById({ id: contact_id })
+      .then((contactObj) => {
+        if (contactObj) {
+          const { phone_number, first_name, last_name } = contactObj.attributes;
+          const name = last_name ? `${first_name} ${last_name}` : first_name;
+          const dialIdParams = { campaignId: campaign_id, userId: user_id, callId: call_id };
+          const nowCallingTwiml = sayDialingContact(name, phone_number, dialIdParams);
+          return res.status(200)
+            .set({ 'Content-Type': 'text/xml' })
+            .send(nowCallingTwiml);
+        }
+        return res.status(404).json({ message: `Could not find contact id: ${contact_id}` });
+      })
+      .catch((err) => {
+        res.status(500).json({ message: `Could not process request to connect volunteer to contact id: ${contact_id}:  ${err}` });
+      });
+  })
+  .catch(err => err);
```

Comment:
Would be nice to return a more explicit error using res in this case too.

Reply:
got it 😄